pandas/tests/indexes/datetimes/test_misc.py

```diff
@@ -191,8 +191,9 @@ def test_datetimeindex_accessors6(self):
         assert [d.weekofyear for d in dates] == expected
 
     # GH 12806
+    # error: Unsupported operand types for + ("List[None]" and "List[str]")
     @pytest.mark.parametrize(
-        "time_locale", [None] if tm.get_locales() is None else [None] + tm.get_locales()
+        "time_locale", [None] + (tm.get_locales() or [])  # type: ignore[operator]
```

Comment:
@twoertwein do you think I would be able to fix this ignore somehow?

I found https://github.com/python/mypy/issues/3933 might be related?

Reply:
I don't think so, would need to use a cast or create the lists not inline and add explicit type annotations.

If `tm.get_locales()` can return None, mypy might accept it if you move the `tm.get_locales()` call before `None`.

Reply:
Reversing the order yields `error: List item 0 has incompatible type "None"; expected "str"  [list-item]`

I think I'll just keep this ignore for now. Adding the casts may make these tests a little more verbose than necessary.